File: 백준/Silver/1966. 프린터 큐/프린터 큐.py
from collections import deque
import sys
n = int(sys.stdin.readline().strip())
for _ in range(n):
    k, m = map(int, sys.stdin.readline().split())
    l1=deque(list(map(int, sys.stdin.readline().strip().split())))

    if k==1 and m==0:
        print(1)
    else:
        cnt=0
        while True:
            first=max(l1)
            now=l1.popleft()
            m-=1
            #print(m)
            #print(first, l1)
            if now==first:
                cnt+=1
                if m<0:
                    break
                #print("pop", l1)
            else:
                l1.append(now)
                if m<0:
                    m+=len(l1)
                    #print(len(l1))
                #print("pop and append", l1)
        print(cnt)

File: 백준/Silver/1966. 프린터 큐/프린터 큐.js
let fs = require('fs');

const inputs = fs.readFileSync('/dev/stdin').toString().trim().split('\n');
//const inputs = fs.readFileSync(__dirname+'/ex2.txt').toString().split('\n');

const k = Number(inputs[0]);


for (let i=1; i < k*2; i+=2){
    let [n, m] = inputs[i].split(' ').map(Number);
    let arr = inputs[i+1].split(' ').map(Number);

    if (n === 1 && m === 0) {
        console.log(1);
    }
    else {
        let cnt = 0;
        while (true) {
            let first = Math.max(...arr);
            let now = arr.shift();
            m--;
            if (now === first) {
                cnt+=1;
                if (m < 0) {
                    break;
                }
            }
            else {
                arr.push(now)
                if (m<0){
                    m+=arr.length;
                }
            }
        }
        console.log(cnt);
    }
}